songwidget.cpp

```diff
@@ -288,6 +293,9 @@ void SongWidget::on_lineEditSearch_textEdited(QString text)
     // If no full-text search is in progress, then filter
     if(!ui->pushButtonClearResults->isVisible())
     {
+        // Sort by song number
+        ui->songs_view->sortByColumn(1,Qt::AscendingOrder);
```

Comment:
I would do the following:


 **// If search test is numeric, sort by the number, esle sort by title
        bool ok;
        text.toInt(&ok);
        if(ok)
        {
            ui->songs_view->sortByColumn(1,Qt::AscendingOrder);
        }
        else
        {
            ui->songs_view->sortByColumn(2,Qt::AscendingOrder);
        }**

This way if the search test is a number, sorting will be done by number, otherwise sorting will be done by title